pkg/storage/tagStore.go

```diff
@@ -0,0 +1,110 @@
+package storage
+
+import (
+	_ "embed"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/google/uuid"
+
+	"github.com/cmsgov/mint-app/pkg/models"
+	"github.com/cmsgov/mint-app/pkg/shared/utilityUUID"
+	"github.com/cmsgov/mint-app/pkg/storage/genericmodel"
+)
+
+//go:embed SQL/tag/create.sql
+var tagCreateSQL string
+
+//go:embed SQL/tag/create_collection.sql
+var tagCreateCollectionSQL string
+
+//go:embed SQL/tag/get_by_table_field_and_content_id.sql
+var tagGetByTableFieldAndContentIDSQL string
+
+// TagCreate writes a new tage to the database
+func (s *Store) TagCreate(
+	logger *zap.Logger,
+	tag *models.Tag,
+) (*models.Tag, error) {
+	tag.ID = utilityUUID.ValueOrNewUUID(tag.ID)
+
+	stmt, err := s.db.PrepareNamed(tagCreateSQL)
+	if err != nil {
+		return nil, genericmodel.HandleModelCreationError(logger, err, tag)
+	}
+	defer stmt.Close()
+
+	retTag := &models.Tag{}
+	err = stmt.Get(retTag, tag)
+	if err != nil {
+		return nil, genericmodel.HandleModelCreationError(logger, err, retTag)
+	}
+
+	return retTag, nil
+}
+
+// TagCollectionCreate creates an array of tags in the database based on the tag contet provided in the string
+func (s *Store) TagCollectionCreate(logger *zap.Logger, tags []*models.Tag, createdBy uuid.UUID) ([]*models.Tag, error) {
+
+	retTags := []*models.Tag{}
+	stmt, sErr := s.db.PrepareNamed(tagCreateCollectionSQL)
+	if sErr != nil {
+		return nil, sErr
+	}
+	defer stmt.Close()
+	errs := []*error{}
+	mapSlice := []map[string]interface{}{}
```

Comment:
This approach is similar to the serialization occurring in the data loader. It also means, it only creates tags if all tags are valid. One invalid tag means none of them will be saved.